projects/plugins/jetpack/modules/videopress/class.videopress-ajax.php

```diff
@@ -41,6 +43,118 @@ public static function init() {
 		return self::$instance;
 	}
 
+	/**
+	 * Ajax method that is used by the VideoPress player to get a token to play a video.
+	 *
+	 * This is used for both logged in and logged out users.
+	 *
+	 * @return void
+	 */
+	public function wp_ajax_videopress_get_playback_jwt() {
+		$guid = filter_input( INPUT_POST, 'guid' );
+		if ( empty( $guid ) ) {
+			wp_send_json_error( array( 'message' => __( 'need a guid', 'jetpack' ) ) );
+			return;
+		}
+
+		if ( ! $this->is_current_user_authed_for_video( $guid ) ) {
+			return;
+		}
+
+		$token = $this->request_jwt_from_wpcom( $guid );
+
+		if ( empty( $token ) ) {
+			wp_send_json_error( array( 'message' => __( 'Could not obtain a VideoPress playback JWT. Please try again later. (empty upload token)', 'jetpack' ) ) );
+			return;
+		}
+
+		if ( is_wp_error( $token ) ) {
+			wp_send_json_error( array( 'message' => __( 'Could not obtain a VideoPress upload JWT. Please try again later.', 'jetpack' ) ) );
+			return;
+		}
+
+		wp_send_json_success( array( 'jwt' => $token ) );
+	}
+
+	/**
+	 * Determines if the current user can view the provided video.
+	 *
+	 * Filterable for 3rd party plugins.
+	 *
+	 * @param string $guid The video id being checked.
+	 */
+	private function is_current_user_authed_for_video( $guid ) {
+		$attachment = videopress_get_post_by_guid( $guid );
+		if ( ! $attachment ) {
+			return false;
+		}
+
+		$video_info = video_get_info_by_blogpostid( get_current_blog_id(), $attachment->ID );
+		if ( null === $video_info->guid ) {
+			return false;
+		}
```

Comment:
Could this be replaced with video_get_info_by_guid( $guid ) ? (no attachment required)

Reply:
This is not available as a JP function unfortunately.